apps/prairielearn/src/pages/home/home.html.ts

```diff
@@ -0,0 +1,260 @@
+
+export const InstructorCourseSchema = z.object({
+  id: CourseSchema.shape.id,
+  label: z.string(),
+  do_link: z.boolean(),
+  course_instances: z.array(
+    z.object({
+      id: CourseInstanceSchema.shape.id,
+      label: CourseInstanceSchema.shape.long_name,
+    }),
+  ),
+});
+export type InstructorCourse = z.infer<typeof InstructorCourseSchema>;
+
+export const StudentCourseSchema = z.object({
+  id: CourseInstanceSchema.shape.id,
+  label: z.string(),
+});
+export type StudentCourse = z.infer<typeof StudentCourseSchema>;
+
+export function Home({
+  resLocals,
+  instructorCourses,
+  studentCourses,
+}: {
+  resLocals: Record<string, any>;
+  instructorCourses: InstructorCourse[];
+  studentCourses: StudentCourse[];
+}) {
+  const { authn_provider_name } = resLocals;
+  return html`
+    <!doctype html>
+    <html lang="en" class="h-100">
+      <head>
+        ${renderEjs(import.meta.url, "<%- include('../partials/head'); %>", resLocals)}
+      </head>
+
+      <body class="d-flex flex-column h-100">
+        <header>
+          ${renderEjs(import.meta.url, "<%- include('../partials/navbar'); %>", {
+            ...resLocals,
+            navPage: 'home',
+          })}
+        </header>
+
+        <main class="flex-grow-1">
+          ${ActionsHeader()}
+
+          <div id="content" class="container py-5">
+            ${DevModeCard()} ${InstructorCoursesCard({ instructorCourses })}
+            ${StudentCoursesCard({
+              studentCourses,
+              hasInstructorCourses: instructorCourses.length > 0,
+              canAddCourses: authn_provider_name !== 'LTI',
+            })}
+          </div>
+        </main>
+
+        ${config.homepageFooterText && config.homepageFooterTextHref
+          ? html`
+              <footer class="footer font-weight-light text-light text-center small">
+                <div class="bg-secondary p-1">
+                  <a class="text-light" href="${config.homepageFooterTextHref}">
+                    ${config.homepageFooterText}
+                  </a>
+                </div>
+              </footer>
+            `
+          : ''}
+      </body>
+    </html>
+  `.toString();
+}
+
+function ActionsHeader() {
+  return html`
+    <div class="container">
+      <div class="row">
+        <div class="col-md-6">
+          <div class="card rounded-pill my-1">
+            <div class="card-body d-flex align-items-center p-2">
+              <span class="fa-stack fa-1x mr-1" aria-hidden="true">
+                <i class="fas fa-circle fa-stack-2x text-secondary"></i>
+                <i class="fas fa-user-graduate fa-stack-1x text-light"></i>
+              </span>
+              <span class="small p-2 font-weight-bold text-uppercase text-secondary">
+                Students
+              </span>
+              <a href="${config.urlPrefix}/enroll" class="btn btn-xs btn-outline-primary">
+                Enroll course
+              </a>
```

Comment:
These were changed as per discussion in https://github.com/PrairieLearn/PrairieLearn/pull/10177#discussion_r1666809274. The visual representation is slightly different but is still roughly the same.

Reply:
LGTM!